client/js/render.js

```diff
@@ -90,6 +99,54 @@ function buildChatMessage(data) {
 	return msg;
 }
 
+function renderPreview(preview, msg) {
```

Comment:
Why exactly do you need it? You already append the preview in template, and you seem to be duplicating it here.